// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package teeapps.params;

message PredictionBiasEvalParams {
  string label_field_name = 1;
  string score_field_name = 2;
  int64 bucket_num = 3;
  int64 min_item_cnt_per_bucket = 4;
  // allowed: {"equal_width", "equal_frequency"}
  // default: "equal_width"
  string bucket_method = 5;
}

message PredictionBiasEvalReport {
  message BucketPredictionBiasEvalReport{
    float left_endpoint = 1;
    bool left_closed = 2;
    float right_endpoint = 3;
    bool right_closed = 4;
    bool is_na = 5;
    float avg_prediction = 6;
    float avg_label = 7;
    float bias = 8;
  }

  repeated BucketPredictionBiasEvalReport bucket_reports = 1;
}